docs/producing-and-consuming.md

```diff
@@ -202,7 +202,7 @@ Because an EOS consumer is difficult to implement correctly, all details have
 been abstracted away to a [`GroupTransactSession`][20] type. See the
 [transactions](./transactions.md) page for more details.
 
-### The cooperative balancer
```

Comment:
I see what you mean about putting these under the consumer group only, but these are important enough concepts that I'd leave them as ### rather than ##### (here and below)

Other changes in this PR look good though